src/App.tsx

```diff
@@ -1,19 +1,171 @@
+      }));
+
+      return;
+    }
+
+    const todoId = () => {
+      if (this.state.addedTodos.length === 0) {
+        return 1;
+      }
+
+      const array = this.state.addedTodos.map(item => item.id);
+      const maxNum = Math.max(...array);
+
+      return maxNum + 1;
+    };
+
+    const addedUser = users.find(user => user.name === this.state.name);
+    const newTodo = {
+      userId: addedUser?.id,
+      id: todoId(),
+      title: this.state.title.trim().slice(0, this.state.maxTitleLength),
+      completed: false,
+      user: addedUser || null,
+    };
+
+    this.setState(state => ({
+      addedTodos: [...state.addedTodos, newTodo],
+      title: '',
+      name: '',
+    }));
+  };
+
+  removeTodo = (id: number) => {
+    this.setState(state => ({
+      addedTodos: [...state.addedTodos].filter(item => item.id !== id),
+    }));
+  };
+
+  render() {
+    const { addedTodos } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Todos list</h1>
+
+        <TodoList preparedTodos={addedTodos} removeTodo={this.removeTodo} />
+
+        <h2>Here you can add you todo to the list</h2>
+
+        <form
+          onSubmit={(event) => {
+            event.preventDefault();
+          }}
+        >
+          <input
+            type="text"
+            name="title"
+            placeholder="Add todo"
+            value={this.state.title}
+            onChange={(event) => {
+              this.setState({ title: event.target.value });
+            }}
+            onFocus={() => {
+              if (this.state.titleError) {
+                this.setState(state => ({
+                  titleError: !state.titleError,
+                }));
+              }
+            }}
+          />
+          <span
+            className={classNames('error', { error__show: this.state.titleError })}
+          >
+            You should add a todo using only letters and digits
+          </span>
+          <select
+            name="name"
+            id="name"
+            value={this.state.name}
+            onChange={(event) => {
+              this.setState({
+                name: event.target.value,
+              });
+            }}
+            onFocus={() => {
+              if (this.state.nameError) {
+                this.setState(state => ({
+                  nameError: !state.nameError,
+                }));
+              }
+            }}
+          >
+            <option value="">Choose a responsible person</option>
+            {users.map(user => (
+              <option
+                key={user.name}
+                value={user.name}
+              >
+                {user.name}
+              </option>
+            ))}
+          </select>
+          <span
+            className={classNames('error', { error__show: this.state.nameError })}
+          >
+            You should add user
+          </span>
+          <button
+            type="submit"
+            onClick={this.addTodo}
```

Comment:
add new todo on form submit, not on button click

Reply:
fixed